src/libxrpl/protocol/STInteger.cpp

```diff
@@ -194,11 +194,25 @@ STUInt64::getText() const
 template <>
 Json::Value STUInt64::getJson(JsonOptions) const
 {
-    std::string str(16, 0);
-    auto ret = std::to_chars(str.data(), str.data() + str.size(), value_, 16);
-    assert(ret.ec == std::errc());
-    str.resize(std::distance(str.data(), ret.ptr));
-    return str;
+    auto convertToString = [](uint64_t const value, int const base) {
+        assert(base == 10 || base == 16);
+        std::string str(
+            base == 10 ? 20 : 16, 0);  // Allocate space depending on base
+        auto ret =
+            std::to_chars(str.data(), str.data() + str.size(), value, base);
+        assert(ret.ec == std::errc());
+        str.resize(std::distance(str.data(), ret.ptr));
+        return str;
+    };
+
+    if (auto const& fName = getFName(); fName == sfMaximumAmount ||
```

Comment:
Doesn't feel right having ST* types having knowledge of specific fields. Is it the only way?

Reply:
STUInt16 has something similar, so i think this is fine
```
Json::Value STUInt16::getJson(JsonOptions) const
{
    if (getFName() == sfLedgerEntryType)
    {
        auto item = LedgerFormats::getInstance().findByType(
            safe_cast<LedgerEntryType>(value_));
        if (item != nullptr)
            return item->getName();
    }
```